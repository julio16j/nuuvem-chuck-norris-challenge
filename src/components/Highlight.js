import React from "react";

function HighlighText ({ text = "", regexTerm = "" }) {
  if (!regexTerm.trim()) {
    return <span>{text}</span>;
  }
  const regex = new RegExp(`(${regexTerm})`, "i");
  const parts = text.split(regex);

  return (
    <span>
      {parts.filter(String).map((part, i) => {
        return regex.test(part) ? (
          <mark key={i}>{part}</mark>
        ) : (
          <span key={i}>{part}</span>
        );
      })}
    </span>
  );
};
export default HighlighText