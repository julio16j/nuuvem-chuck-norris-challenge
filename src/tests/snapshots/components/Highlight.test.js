import renderer from "react-test-renderer";
import Highlight from "../../../components/Highlight";
import React from "react";

it("Highlight Component renders correctly", () => {
  const component = renderer
    .create(<Highlight text={"Test Message"} regexTerm={"Test"} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
