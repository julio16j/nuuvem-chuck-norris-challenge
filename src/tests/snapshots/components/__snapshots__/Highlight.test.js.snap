// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight Component renders correctly 1`] = `
<span>
  <mark>
    Test
  </mark>
  <span>
     Message
  </span>
</span>
`;
